.github/workflows/deploy-docs.yml

```diff
@@ -0,0 +1,39 @@
+name: Deploy Documentation
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: mlugg/setup-zig@v1
+        with:
+          version: 0.14.0
+      - run: zig build docs
+
+      - name: Upload Pages artifact
+        uses: actions/upload-pages-artifact@v3
+        with:
+          path: ./zig-out/docs
+
+  deploy:
+    needs: build
+    runs-on: ubuntu-latest
+
+    permissions:
+      pages: write
+      id-token: write
+
+    environment:
+      name: github-pages
+      url: ${{ steps.deployment.outputs.page_url }}
```

Comment:
What does this do?
Do we need it?

Reply:
Thanks for pointing me to the docs, I admit I am not entirely sure I understand what points 3 to 5 mean.
Do you think we should create an environment in the repo settings?
I don't have prior experience of deploying to GitHub pages 😄